### ISSUE TITLE
TF_SessionRun with multiple inputs gives Segmentation Fault

### ISSUE THREAD

**Issue by Vaishu-Jeeva** [tensorflow/tensorflow#47043]
System information
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux RHEL
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: no
TensorFlow installed from (source or binary): source
TensorFlow version (use command below): 2.4
Python version: 3.6
Bazel version (if compiling from source): 0.16.1
Exact command to reproduce: compile and execute my program

Problem description:
i have created a logistic regression model using tf.estimators.linearClassifier and exported the model to serve it using C API.

the saved_model cli command has the following output:
The given SavedModel SignatureDef contains the following input(s):
inputs['a'] tensor_info:
dtype: DT_INT64
shape: (-1, 1)
name: Placeholder:0
inputs['b'] tensor_info:
dtype: DT_INT64
shape: (-1, 1)
name: Placeholder_1:0
The given SavedModel SignatureDef contains the following output(s):
outputs['all_class_ids'] tensor_info:
dtype: DT_INT32
shape: (-1, 2)
name: head/predictions/Tile:0
outputs['all_classes'] tensor_info:
dtype: DT_STRING
shape: (-1, 2)
name: head/predictions/Tile_1:0
outputs['class_ids'] tensor_info:
dtype: DT_INT64
shape: (-1, 1)
name: head/predictions/ExpandDims:0
outputs['classes'] tensor_info:
dtype: DT_STRING
shape: (-1, 1)
name: head/predictions/str_classes:0
outputs['logistic'] tensor_info:
dtype: DT_FLOAT
shape: (-1, 1)
name: head/predictions/logistic:0
outputs['logits'] tensor_info:
dtype: DT_FLOAT
shape: (-1, 1)
name: linear/linear_model/linear/linear_model/linear/linear_model/weighted_sum:0
outputs['probabilities'] tensor_info:
dtype: DT_FLOAT
shape: (-1, 2)
name: head/predictions/probabilities:0
Method name is: tensorflow/serving/predict

it has two input tensors and 5 outputs, the following is the c code use to run this  model:

#include <stdlib.h>
#include <stdio.h>
#include "tensorflow/c/c_api.h"

void NoOpDeallocator(void* data, size_t a, void* b) {}

int main()
{
    //********* Read model
    TF_Graph* Graph = TF_NewGraph();
    TF_Status* Status = TF_NewStatus();

    TF_SessionOptions* SessionOpts = TF_NewSessionOptions();
    TF_Buffer* RunOpts = NULL;

    const char* saved_model_dir = "/nobackup/rajassub/scripts/C-Programs/tensorflow/logicalRegression/";
    const char* tags = "serve"; // default model serving tag; can change in future
    int ntags = 1;

    TF_Session* Session = TF_LoadSessionFromSavedModel(SessionOpts, RunOpts, saved_model_dir, &tags, ntags, Graph, NULL, Status);
    if(TF_GetCode(Status) == TF_OK)
    {
        printf("TF_LoadSessionFromSavedModel OK\n");
    }
    else
    {
        printf("%s",TF_Message(Status));
    }

    //****** Get input tensor
    //TODO : need to use saved_model_cli to read saved_model arch
    int NumInputs = 2;

    TF_Output* Input = (TF_Output*)malloc(sizeof(TF_Output) * NumInputs);

    TF_Output t0 = {TF_GraphOperationByName(Graph, "Placeholder"), 0};
    if(t0.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName placeholder\n");
    else
	printf("TF_GraphOperationByName placeholder is OK\n");
    
	
    TF_Output t1 = {TF_GraphOperationByName(Graph, "Placeholder_1"), 0};
    if(t1.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName placeholder_1\n");
    else
	printf("TF_GraphOperationByName placeholder_1 is OK\n");
	
    Input[0] = t0;
	Input[1] = t1;
    
    //********* Get Output tensor
    int NumOutputs = 5;
    TF_Output* Output = (TF_Output*)malloc(sizeof(TF_Output) * NumOutputs);

    TF_Output t2 = {TF_GraphOperationByName(Graph, "head/predictions/ExpandDims"), 0};
    if(t2.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName linear/head/predictions/ExpandDims\n");
    else	
	printf("TF_GraphOperationByName linear/head/predictions/ExpandDims is OK\n");
	
	TF_Output t3 = {TF_GraphOperationByName(Graph, "head/predictions/str_classes"), 0};
    if(t3.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName head/predictions/str_classes\n");
    else	
	printf("TF_GraphOperationByName linear/head/predictions/str_classes is OK\n");
	
    TF_Output t4 = {TF_GraphOperationByName(Graph, "head/predictions/logistic"), 0};
    if(t4.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName linear/head/predictions/logistic\n");
    else	
	printf("TF_GraphOperationByName linear/head/predictions/logistic is OK\n");
	
	TF_Output t5 = {TF_GraphOperationByName(Graph, "linear/linear_model/linear/linear_model/linear/linear_model/weighted_sum"), 0};
    if(t5.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName linear/linear_model/weighted_sum\n");
    else	
	printf("TF_GraphOperationByName linear/linear_model/weighted_sum is OK\n");
	
	TF_Output t6 = {TF_GraphOperationByName(Graph, "head/predictions/probabilities"), 0};
    if(t6.oper == NULL)
        printf("ERROR: Failed TF_GraphOperationByName linear/head/predictions/probabilities\n");
    else	
	printf("TF_GraphOperationByName linear/head/predictions/probabilities is OK\n");
	
    Output[0] = t2;
	Output[1] = t3;
	Output[2] = t4;
	Output[3] = t5;
	Output[4] = t6;

    //********* Allocate data for inputs & outputs
    TF_Tensor** InputValues = (TF_Tensor**)malloc(sizeof(TF_Tensor*)*NumInputs);
    TF_Tensor** OutputValues = (TF_Tensor**)malloc(sizeof(TF_Tensor*)*NumOutputs);

    int ndims = 2;
    int64_t dims[] = {1,1};
    int64_t data1[] = {2};

    int ndata = sizeof(int64_t)*1 ;// This is tricky, it number of bytes not number of element

    TF_Tensor* int_tensor1 = TF_NewTensor(TF_FLOAT, dims, ndims, data1, ndata, &NoOpDeallocator, 0);
    if (int_tensor1 != NULL)
    {
        printf("TF_NewTensor is OK for int_tensor1\n");
    }
    else
	printf("ERROR: Failed TF_NewTensor for int_tensor1\n");
    
	int64_t data2[] = {3};
	TF_Tensor* int_tensor2 = TF_NewTensor(TF_FLOAT, dims, ndims, data2, ndata, &NoOpDeallocator, 0);
    if (int_tensor2 != NULL)
    {
        printf("TF_NewTensor is OK for int_tensor2\n");
    }
    else
	printf("ERROR: Failed TF_NewTensor for int_tensor2\n");
    InputValues[0] = int_tensor1;
	InputValues[1] = int_tensor2;
    
    // //Run the Session
    TF_SessionRun(Session, NULL, Input, InputValues, NumInputs, Output, OutputValues, NumOutputs, NULL, 0,NULL , Status);

    if(TF_GetCode(Status) == TF_OK)
    {
        printf("Session is OK\n");
    }
    else
    {
        printf("%s",TF_Message(Status));
    }

    // //Free memory
    TF_DeleteGraph(Graph);
    TF_DeleteSession(Session, Status);
    TF_DeleteSessionOptions(SessionOpts);
    TF_DeleteStatus(Status);

    void* buff = TF_TensorData(OutputValues[0]);
    float* offsets = buff;
    printf("Result Tensor :\n");
    printf("%f\n",offsets[0]);
    return 0; 
    
}

But it is crashing at Session run step, it would be really helpful if someone could identify where the issue is.

output obtained:
TF_LoadSessionFromSavedModel OK
TF_GraphOperationByName placeholder is OK
TF_GraphOperationByName placeholder_1 is OK
TF_GraphOperationByName linear/head/predictions/ExpandDims is OK
TF_GraphOperationByName linear/head/predictions/str_classes is OK
TF_GraphOperationByName linear/head/predictions/logistic is OK
TF_GraphOperationByName linear/linear_model/weighted_sum is OK
TF_GraphOperationByName linear/head/predictions/probabilities is OK
TF_NewTensor is OK for int_tensor1
TF_NewTensor is OK for int_tensor2
Segmentation fault (core dumped)

**Comment by sanatmpa1** [comment]
@Vaishu-Jeeva,

Can you take a look at this similar [issue1](https://github.com/tensorflow/tensorflow/issues/19347), [issue2](https://github.com/tensorflow/tensorflow/issues/2034) which is about segmentation fault? We also recommend you to update the tensorflow to latest stable version and let us know if the issue still persists. Thanks!
